packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -546,19 +546,19 @@ class _CupertinoButtonState extends State<CupertinoButton> with SingleTickerProv
               child: FadeTransition(
                 opacity: _opacityAnimation,
                 child: DecoratedBox(
-                  decoration: BoxDecoration(
-                    border:
-                        enabled && isFocused
-                            ? Border.fromBorderSide(
-                              BorderSide(
+                  decoration: ShapeDecoration(
+                    shape: RoundedSuperellipseBorder(
+                      side:
+                          enabled && isFocused
```

Comment:
Nit: This formatting looks strange (though it was strange before your change). If you manually put it on one line (`side: enabled && isFocused`) does the autoformatter undo it?

Reply:
It sadly does. I've extracted the shapeDecoration to a separate variable, which makes it less bad for being on its own.